draft-ietf-tokbind-protocol-12.xml

```diff
@@ -437,20 +437,23 @@
           </list>
         </t>
 
-        <t>Please note that TLS 1.2 and earlier versions support renegotiaion, which produces a new 
-        TLS master secret for the same connection, with its associated session keys and EKM value.
-        TokenBinding.signature MUST use the EKM value derived from the same TLS master secret that 
-        produced the session keys encrypting the application_data record containing this 
-        TokenBinding. Such use of the current EKM for the TLS connection prevents the replay of 
-        bound tokens within renegotiated TLS sessions, but requires the application to synchronize 
-        Token Binding message generation and verification with the TLS handshake state.</t>
-
-        <t>Specifications defining the use of Token Binding with various application protocols, 
-        such as Token Binding over HTTP <xref target="I-D.ietf-tokbind-https"/>, may prohibit the 
-        use of TLS renegotiation in combination with Token Binding. Alternatively, such 
-        specifications need to define a way to determine which EKM value corresponds to a given 
-        TokenBindingMessage, and a mechanism preventing a TokenBindingMessage from being split 
-        across the renegotiation boundary.</t>
+        <t>Please note that TLS 1.2 and earlier versions support renegotiation, which produces a new
+        TLS master secret for the same connection, with associated session keys and EKM value.
+        TokenBinding.signature MUST be a signature of the EKM value derived from the TLS master
+        secret that produced the session keys encrypting the TLS application_data record containing
+        this TokenBinding. Such use of the current EKM for the TLS connection makes replay of bound
+        tokens within renegotiated TLS sessions detectable, but requires the application to
+        synchronize Token Binding message generation and verification with the TLS handshake
+        state.</t>
+
+        <t>Specifications defining the use of Token Binding with application protocols, such as
+        Token Binding over HTTP <xref target="I-D.ietf-tokbind-https"/>, may prohibit the use of TLS
+        renegotiation in combination with Token Binding, obviating the need for such
+        synchronization. Alternatively, such specifications need to define a way to determine which
+        EKM value corresponds to a given TokenBindingMessage, and a mechanism preventing a
+        TokenBindingMessage from being split across TLS renegotiation boundaries (i.e., due to TLS
+        record layer fragmentation of "client messages" -- see Section 6.2.1 of <xref
+        target="RFC5246"/>).</t>
```

Comment:
I would also note that application-layer messages that carry a token binding might cross the boundary in ways that make processing difficult.  I don't think that we have generic guidance, but your HTTP proposal addresses that concern neatly.